flixel/addons/weapon/FlxWeapon.hx

```diff
@@ -55,72 +56,58 @@ class FlxTypedWeapon<TBullet:FlxBullet>
 	public static inline var BULLET_SOUTH_WEST:Int = 135;
 
 	/**
-	 * Internal name for this weapon (i.e. "pulse rifle")
+	 * Internal name for this weapon (e.g., `"pulse rifle"`).
 	 */
 	public var name:String;
 
 	/**
-	 * The FlxGroup into which all the bullets for this weapon are drawn. This should be added to your display and collision checked against it.
+	 * The `FlxGroup` from which all the bullets for this weapon are drawn. This should be added to your display and collision-checked against it.
 	 */
 	public var group(default, null):FlxTypedGroup<TBullet>;
 
-	// Internal variables, use with caution
-	public var nextFire:Int = 0;
-
 	/**
-	 * The delay in milliseconds (ms) between which each bullet is fired, set to zero to clear. By default there is no rate, as it can be controlled by FlxControl.setFireButton.
-	 * However if you are firing using the mouse you may wish to set a firing rate.
+	 * The delay in milliseconds (ms) between which each bullet is fired. Default is 0, which means there is no delay.
 	 */
 	public var fireRate:Int = 0;
 
 	/**
-	 * When a bullet goes outside of this bounds it will be automatically killed, freeing it up for firing again.
-	 * TODO - Needs testing with a scrolling map (when not using single screen display)
-	 */
-	public var bounds:FlxRect;
-
-	/**
-	 * Only accessible when fireFrom is "PARENT"
+	 * A timer used whenever the weapon is fired, with its duration set to `fireRate`.
 	 */
-	public var parent(default, null):FlxSprite;
+	public var fireTimer:FlxTimer;
```

Comment:
I would just revert this aspect of the PR, I don't like using a FlxTimer for this, it should be a private float.

 I would just rewrite FlxWeapon to extend FlxTypedGroup and do it from the ground up in a public repo until it's solid enough to replace this in the next major version.
 
 Truthfully though, If FlxWeapon was written now I probably wouldn't put it in flixel-addons to begin with, it should just be it's own isolated demo that people can use if they like that implementation. If you do rewrite it in a new repo and it's decent I'd rather just remove this entirely and put yours in [External Tools](https://github.com/HaxeFlixel/flixel/wiki/External-Tools)

Reply:
So do you want me to go back to the old implementation, or to try to do one of these other suggestions what you have listed?